### PR TITLE
Slugify: Replace gosimple/slug with a simple function 

### PR DESCRIPTION
**Pull request by ryantxu**
The current slugify process embeds ~1MB of unicode lookup data in every executable 😱 

This replaces the large library with a simple function based on https://github.com/mozillazg/go-slugify

Existing slugs will not change since they are computed on save, however dashboards that are saved again (or the titles change) may have slugs that change.  Regardless, existing URLs will keep working since the slug is only used as an informational part of the URL

### CODE REVIEW

**Comment by DanCech** [comment]
Looks sane enough to me.  My understanding is that slugs are only generated when things are added or updated so they shouldn't change unless someone also changes the title, but I may be wrong on that.

File: pkg/infra/slugify/slugify.go [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+	}
+)
+
+// Slugify creates a URL safe latin slug for a given value
+func Slugify(value string) string {
+	s := simpleSlugger.Slugify(value)
+	if s == "" {
+		// If the name is only characters outside of the
+		// sluggable characters, use a content hash instead
+		s = uuid.NewV5(uuid.NamespaceOID, value).String()
```
Comment (ryantxu) [inline pkg/infra/slugify/slugify.go:30]: previously we used:
```
base64.RawURLEncoding.EncodeToString([]byte(title))
```
that gets pretty big, so i think this is better -- but I do not feel too strongly

File: pkg/models/dashboards_test.go [hunk 2]
```diff
@@ -72,14 +73,14 @@ func TestSlugifyTitle(t *testing.T) {
 	testCases := map[string]string{
 		"Grafana Play Home": "grafana-play-home",
 		"snöräv-över-ån":    "snorav-over-an",
-		"漢字":                "han-zi",      // Hanzi for hanzi
+		"漢字":                "5ryi5a2X",    // "han-zi",      // Hanzi for hanzi
```
Comment (ryantxu) [inline pkg/models/dashboards_test.go:76]: ^^ example where the new slugs will not be great

I have no sense how valuable this is to grafana users

Comment (DanCech) [reply pkg/models/dashboards_test.go:76]: I doubt it's particularly valuable, weighed against the associated overhead.  If it did become a problem we could relatively easily expose the ability for the user to specify their own slug via dashboard settings.

File: pkg/infra/slugify/slugify.go [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+		} else if !lastCharacterWasInvalid {
+			buffer.WriteRune(s.replaceCharacter)
+			lastCharacterWasInvalid = true
+		}
+	}
+
+	return strings.Trim(buffer.String(), string(s.replaceCharacter))
+}
+
+func getDefaultReplacements() map[rune]string {
```
Comment (nmarrs) [inline pkg/infra/slugify/slugify.go:82]: Might be a little cleaner to move this to another file in same folder

**Review by nmarrs — APPROVED** [review]
LGTM 🐌

File: pkg/infra/slugify/slugify.go [hunk 4]
```diff
@@ -0,0 +1,345 @@
+/*
+ * The MIT License (MIT)
+ *
+ * This file evolved from the MIT licensed: https://github.com/machiel/slugify
```
Comment (mildwonkey) [inline pkg/infra/slugify/slugify.go:4]: Do you think we should include a the copyright portion of machiel's license: https://github.com/machiel/slugify/blob/master/LICENSE? The file is a close enough copy that I  would want to see the "proper" attribution.

**Review by mildwonkey — APPROVED** [review]

**Pull request merged**